### PR TITLE
Protect setup files and avoid publisher trust prompts

### PR DESCRIPTION
**Pull request by AmelBawa-msft**

### CODE REVIEW

File: src/windows-dev-config/bootstrap.ps1 [hunk 1]
```diff
@@ -28,201 +26,225 @@ param(
...
     }
 
-    if ($failures.Count -gt 0) {
-        $details = ($failures | ForEach-Object { "    $_" }) -join [Environment]::NewLine
-        throw "The Calm OS payload in '$Directory' failed Microsoft signature verification:$([Environment]::NewLine)$details$([Environment]::NewLine)Setup was not started. Use -AllowUnsigned only when you intentionally want to run the source copy."
+    foreach ($scope in @('MachinePolicy', 'UserPolicy')) {
+        $policy = Get-ExecutionPolicy -Scope $scope
+        if ($policy -ne 'Undefined') {
+            if ($policy -in @('AllSigned', 'Restricted')) {
+                throw "Organization policy ($scope) requires $policy. Prompt-free setup is unavailable; contact your administrator."
```
Comment (JohnMcPMS) [inline src/windows-dev-config/bootstrap.ps1]: If it would prompt, we won't even let you run it?

**Review by JohnMcPMS — COMMENTED** [review]
Given the lack of support for the memory-based signature check, I think the path is:
1. First run creates directory under install root, copies all scripts there.
2. Changes owner and ACLs to admin only write with everyone read/execute
3. Verifies signature on the entry point script one more time
4. Invokes the entry point script with signature validation elevated
5. Elevated script revalidates signatures for all files
6. Continue with actual work

File: src/windows-dev-config/dev-config.ps1 [hunk 2]
```diff
@@ -13,8 +13,20 @@ param(
 $ErrorActionPreference = 'Stop'
 Set-StrictMode -Version Latest
 
-if (-not $AllowUnsigned -and (Get-ExecutionPolicy) -ne 'AllSigned') {
-    throw 'Signed setup requires AllSigned. Launch PowerShell with -ExecutionPolicy AllSigned; Group Policy may override it. Use -AllowUnsigned only for development.'
+$stepsDir = Join-Path $PSScriptRoot 'steps'
+$securityCode = [IO.File]::ReadAllText((Join-Path $stepsDir '_security.ps1'))
+if (-not $AllowUnsigned) {
+    # Verify and execute the same text to avoid a file-swap race.
+    $signature = Get-AuthenticodeSignature -Content ([Text.Encoding]::Unicode.GetBytes($securityCode)) -SourcePathOrExtension '.ps1'
```
[the remark this answers is not in the document]
Comment (AmelBawa-msft) [reply src/windows-dev-config/dev-config.ps1:20]: Both parameters are supported in Windows PowerShell 5.1. The [documentation’s 7.4 note](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.security/get-authenticodesignature?view=powershell-5.1#-content) concerns encoding: earlier versions require UTF-16LE, which this call already supplies through `[Text.Encoding]::Unicode.GetBytes(...)`. I verified this parameter set on Windows PowerShell 5.1 against a Microsoft-signed script; verification returned `Valid`. Resolving as a false positive.

File: src/windows-dev-config/bootstrap.ps1 [hunk 3]
```diff
@@ -28,201 +26,225 @@ param(
...
-    Write-Host "  Ready in $InstallRoot" -ForegroundColor DarkGray
-
-    $shell = if (Get-Command 'pwsh.exe' -ErrorAction SilentlyContinue) { 'pwsh.exe' } else { 'powershell.exe' }
-    $arguments = @('-NoProfile')
+    $flow = if ($AllowUnsigned) { 'src/windows-dev-config' } else { 'windows-dev-config' }
+    $securityCode = (Invoke-RestMethod -Uri "https://raw.githubusercontent.com/$repo/$Ref/$flow/steps/_security.ps1" -UseBasicParsing -TimeoutSec 60).TrimStart([char]0xFEFF)
     if (-not $AllowUnsigned) {
-        $arguments += '-ExecutionPolicy', 'AllSigned'
+        # Windows PowerShell requires UTF-16LE for in-memory signature verification.
+        $signature = Get-AuthenticodeSignature -Content ([Text.Encoding]::Unicode.GetBytes($securityCode)) -SourcePathOrExtension '.ps1'
```
[the remark this answers is not in the document]
Comment (AmelBawa-msft) [reply src/windows-dev-config/bootstrap.ps1:177]: Both parameters are supported in Windows PowerShell 5.1. The [documentation’s 7.4 note](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.security/get-authenticodesignature?view=powershell-5.1#-content) concerns encoding: earlier versions require UTF-16LE, which this call already supplies through `[Text.Encoding]::Unicode.GetBytes(...)`. I verified this parameter set on Windows PowerShell 5.1 against a Microsoft-signed script; verification returned `Valid`. Resolving as a false positive.

File: src/windows-dev-config/bootstrap.ps1 (hunk 1 above)
Comment (AmelBawa-msft) [reply src/windows-dev-config/bootstrap.ps1]: Agreed. Removed the `AllSigned` rejection so setup can proceed with PowerShell’s normal signature checks and publisher-trust prompts. Organization-enforced `Restricted` still blocks setup.

**Review by ranm-msft — COMMENTED** [review]
Read this end to end against the head commit. The hardening looks solid, and several details are clearly deliberate: rejecting reparse points, walking every ancestor with `DeleteSubdirectoriesAndFiles` in the ancestor mask, creating the root atomically with its ACL instead of create-then-ACL, and having `dev-config.ps1` verify and execute the *same in-memory text* of `_security.ps1` so there is no file-swap window. I also confirmed the earlier `AllSigned` concern is resolved in 8c21cce5 - the pre-rejection is gone, `_elevation.ps1` relaunches with `RemoteSigned`, and only an org-enforced `Restricted` still blocks.

Two things I would like your read on:

1. **An existing install root never gets its ACL reconciled.** `New-DevConfigProtectedDirectory` applies the hardened ACL only inside `if (-not (Test-Path -LiteralPath $Path))`, and `Assert-DevConfigProtectedTree` only proves non-admins *lack* dangerous rights - never that the resume identity *has* traverse/read/execute. So a pre-existing Administrators-only `%ProgramData%\CalmOS` passes every check and the elevated run finishes normally. But the resume task is registered `-RunLevel Limited`, and after the reboot PowerShell has to open `dev-config.ps1` before any in-script elevation runs, so it can fail before it ever reaches UAC or the setup log. Could the existing-root path reconcile a canonical ACL, or otherwise verify effective RX for the resume principal? (This expands the still-open Copilot comment on `_security.ps1`.)

2. **Resume identity under over-the-shoulder elevation.** `Suspend-DevConfigForReboot` reads `[WindowsIdentity]::GetCurrent().Name` while already elevated, so if a standard user starts setup and supplies a *different* administrator's credentials, the `-AtLogOn` trigger is registered for that administrator and will not fire when the original user signs back in. Is that scenario in scope?

**Review by JohnMcPMS — APPROVED** [review]

File: src/windows-dev-config/steps/_security.ps1 [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    $Path = [IO.Path]::GetFullPath($Path).TrimEnd('\')
+    $parent = [IO.Directory]::GetParent($Path)
+    if (-not $parent -or -not $parent.Exists) {
+        throw "The parent directory must already exist: $Path"
+    }
+    for ($ancestor = $parent; $null -ne $ancestor; $ancestor = $ancestor.Parent) {
+        Assert-DevConfigProtectedPath -Path $ancestor.FullName -Ancestor
+    }
+
+    if (-not (Test-Path -LiteralPath $Path)) {
```
[the remark this answers is not in the document]
Comment (AmelBawa-msft) [reply src/windows-dev-config/steps/_security.ps1:95]: This is a valid edge case when an existing setup directory lacks non-elevated read access. Fresh installs grant the required read/execute permissions, so normal installs and reruns with unchanged permissions are unaffected. I’m deferring handling of incompatible existing directories to keep this PR focused.
